### DISCUSSION TITLE
getting direct message text

### DISCUSSION THREAD

**Discussion by DanKeane** [tweepy/tweepy#1474] [Questions] [answered]
Hi, im trying to get the last direct message text. 
Ive found how to get the message id by doing "api.list_direct_messages(1)[0].id" so i dont know why "api.list_direct_messages(1)[0].text" wouldn't work.

Any help appreciated.
cheers

**Comment by Harmon758** [comment, marked as answer]
You'll need to access the `message_create` object within the [Direct Message Event object](https://developer.twitter.com/en/docs/twitter-api/v1/direct-messages/sending-and-receiving/guides/message-create-object) and then the `message_data` object within the `message_create` object. Tweepy only exposes the top level properties as attributes, so the latter will be a key in the dictionary that represents the `message_create` object as JSON. Then, you can use `"text"` as the key for the dictionary that represents the `message_data` object as JSON, e.g.:
```py
direct_messages = api.list_direct_messages()
direct_messages[0].message_create["message_data"]["text"]
```

**Comment by DanKeane** [comment]
Cheers working now.
